src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderIndividualSchedule.kt

```diff
@@ -32,10 +32,21 @@ enum class EventClass {
   COMM,
 }
 
+enum class EventType {
+  APP,
+}
+
 @Entity
 @Table(name = "OFFENDER_IND_SCHEDULES")
-@DiscriminatorColumn(name = "EVENT_CLASS")
-@Inheritance()
+@DiscriminatorFormula(
+  """
+    case
+        when EVENT_CLASS = 'INT_MOV' then 'OffenderAppointment'
+        else 'Unknown'
+    end
+""",
+)
+@Inheritance
 abstract class OffenderIndividualSchedule(
```

Comment:
This will eventually included cases for external movements. Apparently discriminator columns don't work with multiple inheritance which is what we'll need for external movements (e.g. grandparent -> parent -> child).